apps/src/templates/sectionProgress/SectionProgress.jsx

```diff
@@ -164,6 +171,12 @@ class SectionProgress extends Component {
     const levelDataInitialized = scriptData && !isLoadingProgress;
     const linkToOverview = this.getLinkToOverview();
     const lessons = scriptData ? scriptData.stages : [];
+    const summaryStyle =
+      currentView === ViewType.SUMMARY ? styles.show : styles.hide;
+    const detailStyle =
+      currentView === ViewType.DETAIL ? styles.show : styles.hide;
+    const standardsStyle =
+      currentView === ViewType.STANDARDS ? styles.show : styles.hide;
```

Comment:
@davidsbailey is this what you had in mind? Now the `onChange` in the Toggle does the logging and calls `setCurrentView`, which in turn updates the `currentView` state. Performance looks comparable, but I'm not hammering it with lots of students who have a bunch of progress.

Reply:
Yes, this is the optimization I was suggesting would be better. The next step is to remove the existing optimization :-) 

you shouldn't need setState calls any more -- you can use `this.props. currentView` to get the previous view, and `nextProps.currentView` to get the next view, which should be all that you need for your logging. If `this.props. currentView` is not set correctly when your component is first rendered, then you should be able to fix this by dispatching `setCurrentView(ViewType.SUMMARY)` prior to first rendering it.

Regarding testing the performance, yeah, you would probably need a section with lots of progress and students to really notice any slowdown. you could also try throttling your CPU via Performance tab in Chrome debugger. If you don't have an account ready to test this locally, then another idea would be to do a quick test on a section in production which is known to have a lot of progress (I think Mike Harvey has one), and then do a sanity check again after this ships.